db/lnbolt/peerdb.go

```diff
@@ -64,51 +64,46 @@ func (pdb *peerboltdb) GetPeerAddrs() ([]lncore.LnAddr, error) {
 	return addrs, nil
 }
 
-func (pdb *peerboltdb) GetPeerInfo(addr lncore.LnAddr) (*lncore.PeerInfo, error) {
+func (pdb *peerboltdb) GetPeerInfo(addr string) (lncore.PeerInfo, error) {
 
 	var raw []byte
 	var err error
+	var peerInfo lncore.PeerInfo
 
 	if pdb.db == nil {
 		logging.Warnf("PDB.db is nil!")
-		return nil, fmt.Errorf("PDB.db is nil")
+		return peerInfo, fmt.Errorf("PDB.db is nil")
```

Comment:
This doesn't really make sense to me because you're still returning *a* peerinfo, but it's invalid.  Sure if `err` is non-nil then you should ignore it anyways but if you mess up and forget then you're going to run into issues if the peerinfo isn't a null pointer.

Reply:
While calling this function, as always is general practice in go, you catch the error first before using the returned `peerInfo` struct. This doesn't change that behaviour and we do this in other places as well.